pkg/compactor/block_upload.go

```diff
@@ -365,7 +366,7 @@ func (c *MultitenantCompactor) validateAndCompleteBlockUpload(logger log.Logger,
 			return
 		}
 
-		cancel()
+		cancel(cancellation.NewErrorf("validation complete"))
```

Comment:
Is it valid to do `cancel(nil)`? I know this sounds weird just like `panic(nil)`, but cancelling something with an error saying "validation complete" sounds like the "task failed successfully" meme.

Reply:
I agree the fact that the carrier must be an error is confusing, but that's what `context.CancelCauseFunc` takes. So basically every time you use a context cancellation as a normal flow of events you're forced to err. But at the same time I think having an explicit reason is better than a vague "context cancelled" string.

Reply:
Thinking twice about this, I think the message should be more aggressive: "validation was already completed" or so: if someone's still waiting on the context at this point, that's an error.

Reply:
I've changed it to `validation completed` in db107b9.